src/crypto/index.ts

```diff
@@ -2897,16 +2897,16 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
      * been changed.
      *
      * @param syncData - Object containing sync tokens associated with this sync
-     * @param syncDeviceLists - device_lists field from /sync, or response from
+     * @param deviceLists - device_lists field from /sync, or response from
      * /keys/changes
      */
-    public async handleDeviceListChanges(
+    public async processDeviceLists(
         syncData: ISyncStateData,
-        syncDeviceLists: Required<ISyncResponse>["device_lists"],
+        deviceLists?: Required<ISyncResponse>["device_lists"],
     ): Promise<void> {
         // Initial syncs don't have device change lists. We'll either get the complete list
         // of changes for the interval or will have invalidated everything in willProcessSync
-        if (!syncData.oldSyncToken) return;
+        if (!syncData.oldSyncToken || !deviceLists) return;
```

Comment:
Suggest we move the check on whether `deviceLists` is defined or not into `sync.ts` and `sliding-sync-sdk.ts`: it simplifies the interface for this API.

Reply:
How moving the check on the upper level simplifies the interface in this case ? The `deviceLists` is still needed in the method signature